drivers/hwinfo/Kconfig

```diff
@@ -139,7 +139,7 @@ config HWINFO_PSOC6
 config HWINFO_GECKO
 	bool "GECKO hwinfo"
 	default y
-	depends on SOC_FAMILY_EXX32 && !SOC_SERIES_EFR32MG21
+	depends on DT_HAS_SILABS_GECKO_HWINFO_ENABLED
```

Comment:
I'm not sure if we should use the device-tree for this. hwinfo has no device instances, no register address, ...
@mbolivar-nordic do you know of any driver where we use the device-tree to only enable it?

Reply:
Hi @alexanderwachter, there is another hwinfo driver enabled with devicetree [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/hwinfo/Kconfig#L70). Also some time earlier in this PR @gmarull suggested to use [DT_HAS_* defines](https://github.com/zephyrproject-rtos/zephyr/pull/50352/commits/cbe8bd62ed4e77b563c289453cffbcf536025363#r991082288). However, if that approach is not acceptable we can change it back to:
```
depends on SOC_FAMILY_EXX32 && !SOC_SERIES_EFR32MG21 && !SOC_SERIES_EFR32BG22
```

Reply:
I just want feedback from the device-tree experts. I'm ok with both.

Reply:
Looks like driver is using RMU (Reset Management Unit?), but all hw info is taken directly from HAL, bypassing DT. I guess we need to use SOC_SERIES* in this case, and use a non-DT device as well. Yet another example where HALs do more harm than good for the consistency of the project.

Reply:
Removed commits https://github.com/zephyrproject-rtos/zephyr/pull/50352/commits/33980eade4ed98bdc9ea3263a5bc162d9460769a and https://github.com/zephyrproject-rtos/zephyr/pull/50352/commits/93075ea6a47791e3c5d20ac7ebf848e66ad518cc so that we don't use devicetree for hwinfo driver.